### ISSUE TITLE
The training loss behaves unexpectedly.

### ISSUE THREAD

**Issue by cl-liamchiu** [ljy0ustc/LLaRA#17]
I followed the README to run the training script. The only difference between my implementation and the original is that I used an A5000 24GB GPU and set the batch size to 2 due to memory constraints.

However, the training loss seems unusual. It starts very low, then sharply increases at one point. After that, the loss doesn't gradually decrease as expected but instead remains consistently high. This behavior doesn't align with my expectations.

I'm not sure if this behavior indicates an issue with my setup or training process. Could you provide any insights on what might be going wrong?

You can see the graph of the training loss here.
![image](https://github.com/user-attachments/assets/e249403c-bd7f-4582-95ea-8be4cafcf663)

**Comment by ljy0ustc** [comment]
Thanks for your attention to our work. 
As you mentioned, batch size and precision will affect the stability of training.

**Comment by zzzqqqyyy** [comment]
I've encountered the same problem with a single A800 80GB GPU, have you solved it yet?

**Comment by cl-liamchiu** [comment]
> I've encountered the same problem with a single A800 80GB GPU, have you solved it yet?

No, I haven't solved it yet. I checked the code and data, but I didn't find any bugs. However, I do have a warning that says:
```bash!
Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration, which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator())
```
I'm not sure if this warning is related to the problem.

Do you have any thoughts on what might be going wrong?

**Comment by cl-liamchiu** [comment]
> I've encountered the same problem with a single A800 80GB GPU, have you solved it yet?

I found that the issue might be related to the learning rate and batch size. I think that a smaller batch size might require a smaller learning rate, and vice versa. You can adjust the learning rate to a suitable value, or simply increase the accumulate_grad_batches value if you decrease the batch size. For me, setting `accumulate_grad_batches=64` and `batch size=2` worked well, and the multiplied result is the same as the original configuration (`accumulate_grad_batches=16`, `batch size=8`).
